src/resonate/dst/scheduler.py

```diff
@@ -282,7 +295,10 @@ def _add_function_to_runnables(
     def _route_fn_or_coroutine(
         self, ctx: Context, promise: Promise[Any], fn_or_coroutine: FnOrCoroutine
     ) -> None:
-        if isgeneratorfunction(fn_or_coroutine.exec_unit):
+        if promise.done():
```

Comment:
If the promise is done when passed to `_route_fn_or_coroutine` then I immediately unblock coros waiting for that promise